db/migrate/20180906225330_create_books.rb

```diff
@@ -0,0 +1,14 @@
+class CreateBooks < ActiveRecord::Migration[5.1]
+  def change
+    create_table :books do |t|
+      t.string :genre, null: false, default: ""
```

Comment:
`default` should not be here, as not specifying any value for model creation will lead to this default, which is indeed a valid not null string, i.e., this `default` is bypassing the `null: false` validation in some way.
You can check this by running `Book.new` (without any arguments) in your rails console.

Reply:
Same for all the lines below.